doc/source/development.rst

```diff
@@ -193,143 +201,242 @@ Make and commit your changes.
 
-.. code:: bash
+.. code-block:: shell-session
 
     $ tox
 
-Once all is done push your local branch to the forked repository
-and head out to GitHub for creating a pull request into the upstream
-repository.
+Once everything is done push your local branch to the forked repository and
+create a pull request into the upstream repository.
 
-.. code:: bash
+.. code-block:: shell-session
 
     $ git push origin my-topic-branch
 
-Thanks much for contributing to KIWI. Your time and work effort is
-very much appreciated.
+Thank you much for contributing to KIWI. Your time and work effort is very
+much appreciated!
 
-Good to know
+
+Coding Style
 ------------
 
-The following sections provides further information about repository
-integrity, version, package and documentation management and are a good
-read to complete the picture of how the KIWI project works.
+KIWI follows the general PEP8 guidelines with the following exceptions:
+
+- We do not use free functions at all. Even utility functions must be part
+  of a class, but should be either prefixed with the `@classmethod` or
+  `@staticmethod` decorators (whichever is more appropriate).
+
+- Constants are not named with all-caps.
+
+- Do not set module and class level constants/variables, put these into the
+  classes' `__init__` method.
+
+- We do not use type hints (yet) as the current code base maintains Python
+  2 compatibility.
+
+
+Documentation
+~~~~~~~~~~~~~
+
+KIWI uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_ for the API and
+user documentation.
+
+In order to build the html documentation call:
+
+.. code-block:: shell-session
+
+    tox -e doc
+
+or to build the full documentation (including a PDF generated by LaTeX
+[#f3]_):
+
+.. code-block:: shell-session
+
+    tox -e packagedoc
+
+
+Document all your classes, methods, their parameters and their types using
+the standard `reStructuredText
+<https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html>`_
+syntax as supported by Sphinx, an example class is documented as follows:
+
+.. code-block:: python
+
+   class Example(object):
+       """
+       **Example class**
+
+       :param str param: A parameter
+       :param bool : Source file name to compress
+       :param list supported_zipper: List of supported compression tools
+       :attr Optional[str] attr: A class attribute
+       """
+       def __init__(self, param, param_w_default=False):
+           self.attr = param if param_w_default else None
+
+       def method(self, param):
+           """
+           A method that takes a parameter.
+
+           :param list param: a parameter
+           :return: whether param is very long
+           :rtype: bool
+           """
+           return len(parame) > 50
+
+
+Try to stick to the following guidelines when documenting source code:
+
+- Classes should be documented directly in their main docstring and not in
+  `__init__`.
+
+- Document **every** parameter including their types.
```

Comment:
Parameter may be a bit vague. You are probably referring to attributes, right? In this special case, public attributes?

```suggestion
- Document **every** public attribute including their types.
```

Reply:
I was actually thinking about function parameters, but public attributes should also be documented.